### ISSUE TITLE
Assets up-scaling (future request)

### ISSUE THREAD

**Issue by baszczewski** [diasurgical/DevilutionX#3]
Please support modern resolutions and post processing filters for up-scaling. This will allow to presents game in modern, but maybe little cartoonish way. I made some samples with *Waifu2x* filter:
![inventory-after](https://user-images.githubusercontent.com/3357435/46398163-a6597300-c6f4-11e8-9928-38ed42f08be5.jpg)
![dungeon-after](https://user-images.githubusercontent.com/3357435/46398172-ae191780-c6f4-11e8-8f28-c5fe0f9982f5.jpg)
![diablo-after](https://user-images.githubusercontent.com/3357435/46398182-b2ddcb80-c6f4-11e8-8594-74596630ac8a.jpg)

**Comment by ApertureSecurity** [comment]
I plan on adding this in.

It shouldn't be too hard either.

Thanks for issue.

**Comment by ApertureSecurity** [comment]
You can now resize the window after game start

**Comment by AJenbo** [comment]
At the moment we have it using a virtual resolution so it's not locked to anything specific. Tee scaling is very basic though.

**Comment by AJenbo** [comment]
The game now runs in fullscreen (native desktop resolution) and uses bilinear scalling.

Waifu2x appears to not be perfomant enough to be used for post processing and processing the assets would require a rewrite of the render engine. hqx might be a more realistic route to go. Implementing it here should be fairly straight forward:
https://github.com/diasurgical/devilutionX/blob/master/Stub/dx.cpp#L83 though you would probably have to disable logical LogicalSize and handle scaling of coordinates manually.
https://github.com/diasurgical/devilutionX/blob/master/Stub/DiabloUI/sdlrender.cpp#L203

https://github.com/diasurgical/devilutionX/blob/master/Stub/DiabloUI/sdlrender.cpp#L405

**Comment by sheepo99** [comment]
I have a question. Will advanced features such as HQ resolutions and other advanced features be supported by the DevilutionX build, or will they be exclusive to the future Devilution Plus Plus package?

**Comment by AJenbo** [comment]
> I have a question. Will advanced features such as HQ resolutions and other advanced features be supported by the DevilutionX build, or will they be exclusive to the future Devilution Plus Plus package?

The groundworks are being laid, the very first step of making the graphics more configurable is basically ready #44. The render is being cleaned up (this is also a huge step toward 64bit support), this will give us the opportunity for making it hardware accelerated and resolution independent. So it may take a bit of time, but it is on it's way.

**Comment by AJenbo** [comment]
Adding Waifu2x asssets depends on #46, as well as allowing for different sprite sizes.

**Comment by ivan89el** [comment]
@baszczewski Tried topaz ai gigapixel turned out even more cute:
![med](https://user-images.githubusercontent.com/35658450/60962844-aee87500-a328-11e9-9739-81afbb193a0d.png)

**Comment by ApertureSecurity** [comment]
@ivan89el This issue is closed. 

I like though the Russian in this. How much of the game is translated into Russian? Can we talk about this in our discord?

**Comment by ivan89el** [comment]
@ApertureSecurity The exe translation was done in hex and the old (1998) translated MPQ resources were used.

**Comment by AJenbo** [comment]
This one was pointed out on the GOG forums:
https://github.com/atheros/xbrzscale

https://i106.fastpic.ru/big/2019/0630/4b/_8c67722d7e40011d9d42cc301f88c64b.png?noht=1

**Comment by qndel** [comment]
Yeah but the thing is none of these are fast enough to be applied in realtime and modifying a screenshot is  not a big deal

**Comment by AJenbo** [comment]
@qndel we could implement this as part of the loading routine, of cause that we rewrite the render to support upscaled tile, but that's on the road map anyway.

**Comment by BF3000** [comment]
What can I say, there is more than enough scaling algorithms to do the job.
Bilinear,Bicubic, Lanczos, Blackman, Spline.
Something super heavy, like Nnedi3 or Waifu2x can be used to scale sprites during level loading, but not realtime rendering.

**Comment by dcworldwide** [comment]
Interesting to note, ESRGAN project results are promising. 

https://github.com/xinntao/ESRGAN
https://www.youtube.com/watch?v=zpwG2uNWhiQ
https://www.youtube.com/watch?v=FsoyC9j9MV0

**Comment by dcworldwide** [comment]
5 years on since op.... Has the technology's scalability improved enough to make this feasible?
